include/awkward/Content.h

```diff
@@ -675,6 +675,43 @@ namespace awkward {
                   bool mask,
                   bool keepdims) const = 0;
 
+    /// @brief This array with one axis removed by applying a Reducer
+    ///
+    /// The user's entry point for this operation is #sort.
+    ///
+    /// @param negaxis The negative axis: `-axis`. That is, `negaxis = 1`
+    /// means the deepest axis level.
+    /// @param starts Staring positions of each group to combine as an
+    /// {@link IndexOf Index}. These are downward pointers from an outer
+    /// structure into this structure with the same meaning as in
+    /// {@link ListArrayOf ListArray}.
+    /// @param parents Groups to combine as an {@link IndexOf Index} of
+    /// upward pointers from this structure to the outer structure to reduce.
+    /// @param outlength The length of the array, after the operation
+    /// completes.
+    /// @param ascending If `true`, the values will be sorted in an ascending
+    /// order.
+    /// @param stable If `true`, the values will be sorted by a
+    /// stable sort algorithm to maintain the relative order of records with
+    /// equal keys (i.e. values).
+    virtual const ContentPtr
+      sort_next(int64_t negaxis,
+                const Index64& starts,
+                const Index64& parents,
+                int64_t outlength,
+                bool ascending,
+                bool stable,
+                bool keepdims) const = 0;
```

Comment:
`keepdims` only makes sense for reducing, not for sorting. It places the result of a summation, for example, inside of a dimension of length 1 instead of removing a dimension, as reducers normally do. (That makes it easier to use reduced arrays in the same expressions with non-reduced arrays; in particular, it allows for broadcasting.)

Sorting maintains the number of dimensions and even the lengths of each dimension, so `keepdims` can't be doing anything. If it's just being passed down the recursion, you can remove it.

Reply:
`keepdims` is used in the `ListOffsetArray` to sort the content and keep its shape. I'll see if there is a simpler way of doing it.

Reply:
At the entry point for `sort_next` and `argsort_next`, the value of `keepdims` is always `true`. It doesn't need to be an argument.